tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/service/impl/CurriculumMembershipServiceTest.java

```diff
@@ -107,4 +115,119 @@ void shouldThrowExceptionWhenPmNotFound() {
     assertThrows(NoSuchElementException.class,
         () -> cmService.save(dto));
   }
+
+  @Test
+  void shouldPatchCmDto() throws MethodArgumentNotValidException, NoSuchMethodException {
+    CurriculumMembershipDTO dto = createDto(CURRICULUM_ID, PM_UUID, START_DATE, END_DATE);
+    dto.setId(CURRICULUM_ID);
+
+    CurriculumMembership cm = new CurriculumMembership();
+    cm.setId(CURRICULUM_ID);
+    cm.setCurriculumStartDate(START_DATE);
+    cm.setCurriculumEndDate(END_DATE);
+    cm.setCurriculumId(CURRICULUM_ID);
+
+    ProgrammeMembership pm = new ProgrammeMembership();
+    pm.setUuid(PM_UUID);
+
+    CurriculumMembership returnedCm = new CurriculumMembership();
+    returnedCm.setId(CURRICULUM_ID);
+    returnedCm.setCurriculumStartDate(START_DATE);
+    returnedCm.setCurriculumEndDate(END_DATE);
+    returnedCm.setCurriculumId(CURRICULUM_ID);
+    returnedCm.setId(CM_ID);
+    returnedCm.setProgrammeMembership(pm);
+
+    CurriculumMembershipDTO returnedCmDto = createDto(CURRICULUM_ID, PM_UUID, START_DATE, END_DATE);
+    returnedCmDto.setId(CM_ID);
+
+    when(cmRepository.findById(dto.getId())).thenReturn(Optional.of(returnedCm));
+
+    when(cmRepository.save(any(CurriculumMembership.class))).thenReturn(returnedCm);
+    when(cmMapper.curriculumMembershipToCurriculumMembershipDto(returnedCm)).thenReturn(
+        returnedCmDto);
+
+    CurriculumMembershipDTO result = cmService.patch(dto);
```

Comment:
The startDate & endDate you use to patch are the same with the startDate & endDate from the DB. This doesn't look sensible to me.